dipy/tracking/streamline.py

```diff
@@ -260,26 +260,68 @@ def select_by_rois(streamlines, rois, include, mode=None, affine=None,
             yield sl
 
 
-def orient_by_rois(streamlines, roi1, roi2, affine=None, copy=True):
+def _orient_generator(out, roi1, roi2):
+    """
+    Helper function to `orient_by_rois`
+
+    Performs the inner loop separately. This is needed, because functions with
+    `yield` always return a generator
+    """
+    for idx, sl in enumerate(out):
+        dist1 = cdist(sl, roi1, 'euclidean')
+        dist2 = cdist(sl, roi2, 'euclidean')
+        min1 = np.argmin(dist1, 0)
+        min2 = np.argmin(dist2, 0)
+        if min1[0] > min2[0]:
+            yield sl[::-1]
+        else:
+            yield sl
+
+
+def _orient_list(out, roi1, roi2):
+    """
+    Helper function to `orient_by_rois`
+
+    Performs the inner loop separately. This is needed, because functions with
+    `yield` always return a generator.
```

Comment:
Maybe mention this function flips streamlines in-place and returns a reference to the updated list?